### ISSUE TITLE
Bad schema return type when created a generic wrapper class for response entity

### ISSUE THREAD

**Issue by kamilkrzywanski** [springdoc/springdoc-openapi#2733]
[project.zip](https://github.com/user-attachments/files/17179820/project.zip)
Bug Description
When I created a wrapper for ResponseEntity with a generic class inside, the API is generated with the wrapper type instead of the expected ResponseEntity type.

Steps to Reproduce

Create a custom wrapper for ResponseEntity with a generic class.
Generate API documentation (e.g., Swagger/OpenAPI).
Notice that the API uses the custom wrapper class instead of ResponseEntity.
Included a zip file with the code.

Expected Behavior
The API should return the exact type wrapped in the ResponseEntity, not the wrapper itself.

**Screenshots**
![image](https://github.com/user-attachments/assets/430c7493-27cd-4d6a-87d3-4d32f73e6a80)
![image](https://github.com/user-attachments/assets/7f880bdc-3b25-458f-b7a9-a2e8bdca401f)
![image](https://github.com/user-attachments/assets/0f398df8-bc9f-4729-8eb5-28086925fd68)

**Additional context**
The bug has already been fixed. I'll write tests and submit a PR shortly!

**Comment by bnasslahsen** [comment]
@kamilkrzywanski,

I have changed the fix, as it breaks other tests.
Do not hesitate to test the new latest SNAPSHOT.

**Comment by kamilkrzywanski** [comment]
You're right, I only checked my test and had hope that this project would have some pipeline.
I've reviewed your changes, and this approach works as long as we have only one wrapper. However, if we enclose it with another class, my test will fail again. I believe the issue mainly lies with ResponseEntity. My second mistake is that the excluded classes list contains interfaces, and we never get the exact class.

This approach pass all tests, and works in my opinion correct:
https://github.com/springdoc/springdoc-openapi/pull/2736
